caravel/assets/javascripts/explorev2/actions/exploreActions.js

```diff
@@ -19,89 +14,52 @@ export const CLEAR_ALL_OPTS = 'CLEAR_ALL_OPTS';
 export const SET_DATASOURCE_TYPE = 'SET_DATASOURCE_TYPE';
 export const SET_FIELD_VALUE = 'SET_FIELD_VALUE';
 
-export function setTimeColumnOpts(timeColumnOpts) {
-  return { type: SET_TIME_COLUMN_OPTS, timeColumnOpts };
+export function setFieldOptions(options) {
+  return { type: SET_FIELD_OPTIONS, options };
 }
 
-export function setTimeGrainOpts(timeGrainOpts) {
-  return { type: SET_TIME_GRAIN_OPTS, timeGrainOpts };
-}
-
-export function setGroupByColumnOpts(groupByColumnOpts) {
-  return { type: SET_GROUPBY_COLUMN_OPTS, groupByColumnOpts };
-}
-
-export function setMetricsOpts(metricsOpts) {
-  return { type: SET_METRICS_OPTS, metricsOpts };
+export function clearAllOpts() {
+  return { type: CLEAR_ALL_OPTS };
 }
 
-export function setColumnOpts(columnOpts) {
-  return { type: SET_COLUMN_OPTS, columnOpts };
+export function setDatasourceType(datasourceType) {
```

Comment:
i don't think we actually need this. we can change a datasource but there is not a concept of changing the datasource type only. datasource type is tied to the datasource. i think this shows up like this because of some white space diffing. we can remove in a follow up pr.